shoop_tests/admin/test_order_creator.py

```diff
@@ -125,14 +143,44 @@ def test_order_creator_valid(rf, admin_user):
             assert line.taxful_price.amount > line.taxless_price.amount
 
 
+def test_purchase_order_creator_valid(rf, admin_user):
+    get_initial_order_status()  # Needed for the API
+    manufacturer = create_random_manufacturer()
+    contact = get_person_contact(admin_user)
+    supplier = get_simple_supplier()
+    state = get_frontend_order_state(type="purchase", manufacturer=manufacturer)
+    order = get_order_from_state(
+        state,
+        admin_user,
+        PurchaseOrderEditView
+    )
+    product_line = filter(lambda line: line["type"] == "product", state["lines"])[0]
```

Comment:
As you probably noticed from the tests, this does not work for Python 3, since filter doesn't return a list with Py3 but rather just an iterable. Use `first(filter(...))` (`from shoop.utils.iterable import first`) or `list(filter(...))[0]`.